### PR TITLE
chore: Remove useless event

### PR DESCRIPTION
**Pull request by ruanxin**
This PR is a follow up to https://github.com/kyma-project/lifecycle-manager/issues/1352, remove some useless event which brings not so much value and reduce traffic cost to API server.

### CODE REVIEW

File: internal/controller/kyma_controller.go [hunk 1]
```diff
@@ -426,35 +411,28 @@ func (r *KymaReconciler) handleDeletingState(ctx context.Context, kyma *v1beta2.
 
 	if r.WatcherEnabled(kyma) {
 		if err := r.SKRWebhookManager.Remove(ctx, kyma); err != nil {
-			// error is expected, try again
-			r.enqueueNormalEvent(kyma, webhookChartRemoval, err.Error())
-			return ctrl.Result{RequeueAfter: r.RequeueIntervals.Busy}, nil
```
Comment (lindnerby) [inline internal/controller/kyma_controller.go:431]: Here the busy interval requeue gets replaced with immediate requeue. What's the reasoning behind?

File: internal/controller/kyma_controller.go [hunk 2]
```diff
@@ -426,35 +411,28 @@ func (r *KymaReconciler) handleDeletingState(ctx context.Context, kyma *v1beta2.
...
+			return r.requeueWithError(ctx, kyma, err)
 		}
 
 		r.RemoteClientCache.Delete(client.ObjectKeyFromObject(kyma))
 		if err := remote.RemoveFinalizersFromRemoteKyma(ctx, r.RemoteSyncNamespace); client.IgnoreNotFound(err) != nil {
-			err = fmt.Errorf("error while trying to remove finalizer from remote: %w", err)
-			r.enqueueWarningEvent(kyma, deletionError, err)
 			r.Metrics.RecordRequeueReason(metrics.FinalizersRemovalFromRemoteKyma, queue.UnexpectedRequeue)
-			return ctrl.Result{}, err
+			return r.requeueWithError(ctx, kyma, err)
```
Comment (lindnerby) [inline internal/controller/kyma_controller.go:429]: Here and in the line 423 you are introducing updateStatus calls now, which weren't required before. Is this expected?

File: internal/declarative/v2/reconciler.go [hunk 3]
```diff
@@ -469,26 +458,19 @@ func (r *Reconciler) renderTargetResources(
...
-			r.Event(obj, "Warning", "PostRenderTransform", err.Error())
 			obj.SetStatus(status.WithState(shared.StateError).WithErr(err))
 			return nil, err
 		}
 	}
 
 	target, err := converter.UnstructuredToInfos(targetResources.Items)
 	if err != nil {
-		// Prevent ETCD load bursts during secret rotation
-		if !util.IsConnectionRelatedError(err) {
```
Comment (lindnerby) [inline internal/declarative/v2/reconciler.go:488]: Is this util function still used somewhere else now?

Comment (ruanxin) [reply internal/declarative/v2/reconciler.go:488]: yes, it's a critical function actually.

File: internal/controller/kyma_controller.go (hunk 2 above)
Comment (ruanxin) [reply internal/controller/kyma_controller.go:429]: yes, when this situation happens, it helps us know the reason why this kyma stuck under deletion by check lastoperation field.

File: internal/controller/kyma_controller.go (hunk 1 above)
Comment (ruanxin) [reply internal/controller/kyma_controller.go:431]: to compliance with requeue standard, when err occurs, requeue with error, there is no needs to control requeue interval for error, it's controlled by rate limit.

**Review by lindnerby — APPROVED** [review]

**Pull request merged**